qt/applications/workbench/workbench/utils/recentlyclosedscriptsmenu.py

```diff
@@ -0,0 +1,98 @@
+# Mantid Repository : https://github.com/mantidproject/mantid
+#
+# Copyright &copy; 2020 ISIS Rutherford Appleton Laboratory UKRI,
+#   NScD Oak Ridge National Laboratory, European Spallation Source,
+#   Institut Laue - Langevin & CSNS, Institute of High Energy Physics, CAS
+# SPDX - License - Identifier: GPL - 3.0 +
+#  This file is part of the mantid workbench.
+import functools
+
+from qtpy.QtWidgets import QMenu, QMessageBox
+from os.path import join, exists
+from mantid import ConfigService
+from mantidqt.utils.qt import create_action
+from config import CONF
+
+RECENT_SCRIPT_MAX_NUMBER = 10
+CACHE_FILE_NAME = "recent_script_file"
+MENU_CACHE_LOCATION = join(ConfigService.getAppDataDirectory(), CACHE_FILE_NAME)
+RECENT_SCRIPTS_KEY = "RecentScripts"
+
+
+class RecentlyClosedScriptsMenu(QMenu):
+    def __init__(self, mainwindow):
+        super(RecentlyClosedScriptsMenu, self).__init__()
+        self.setTitle("Open Recently Closed Scripts")
+        self.aboutToShow.connect(self.repopulate_menu)
+        self.mainwindow = mainwindow
+
+    def repopulate_menu(self):
+        self.clear()
+        self.populate_menu()
+
+    def populate_menu(self):
+        # Check cache is present or don't do anything.
+        scripts = self.load_scripts_from_settings()
+
+        if len(scripts) > 0:
+            for script_path in scripts:
+                script_name = self.size_path_correctly(script_path)
+                new_action = create_action(parent=self.mainwindow,
+                                           text=script_name,
+                                           on_triggered=functools.partial(self.open_script, script_path))
+                self.addAction(new_action)
+        else:
+            self.addAction(create_action(parent=self.mainwindow,
+                                         text="No recently closed scripts found"))
+
+    @staticmethod
+    def size_path_correctly(path):
+        return path if len(path) < 33 else "..." + path[30:]
```

Comment:
Is it worth changing the last param to `path[-30:]` to ensure that you always get the last 30 chars of the string, rather than however many it happens to be without the first 30?

Reply:
Yes it is, good catch!